### DISCUSSION TITLE
Is the instrumentation name exposed as an attribute?

### DISCUSSION THREAD

**Discussion by alvinhom** [open-telemetry/opentelemetry-java-instrumentation#6375] [Q&A]
I am putting the traces into a database so that it is queryable and do analytics.  I would like to group the spans using instrumentation name/type.  I would like to group filter spans containing database instrumentation versus http instrumentation, for example.

I do see the following resource attributes being generated:

      "telemetry.sdk.version": "1.14.0",
      "telemetry.sdk.language": "java",
      "telemetry.sdk.name": "opentelemetry",

I would like to know if the instrumentation name or type is also exposed as an attribute.

**Comment by mateuszrzeszutek** [comment]
Hey @alvinhom ,
What exporter are you using? Depending on the data format that you're receiving, the instrumentation name might be present at different places. For example, OTLP has a dedicated [`InstrumentationScope`](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto#L68) element that contains this data; the Jaeger gRPC exporter puts this information into [span tags](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/jaeger/src/main/java/io/opentelemetry/exporter/jaeger/SpanMarshaler.java#L107), and so does [Zipkin](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/zipkin/src/main/java/io/opentelemetry/exporter/zipkin/ZipkinSpanExporter.java#L135).
